code/modules/shuttle/supply.dm

```diff
@@ -746,6 +752,85 @@
 
 	frequency.post_signal(src, status_signal)
 
+/obj/machinery/computer/supplycomp/proc/sort_and_send_messages()
+	var/list/orders_by_users = list()
+	var/ordernumbers
+	var/recipient
+	var/object_name
+	var/multiple = FALSE
+	for(var/A in SSshuttle.requestlist)
+		var/datum/supply_order/SO = A
+		if(!orders_by_users["[SO.orderedby][SO.object.name]"])         //orders_by_users is an associative list where index is "[requester name][object name]"
+			orders_by_users["[SO.orderedby][SO.object.name]"] = list() //and the value is another list, which contains order datums from same person of same object
+		orders_by_users["[SO.orderedby][SO.object.name]"] += SO        //they are used later to fill the messages with multiple orders
+
+	for(var/A in orders_by_users)                   //take a list from orders_by_users
+		var/list = A
```

Comment:
naming a variable `list` is misleading if it's not a list. Best to rename it to `key` or `requesterName` or the like.

Reply:
done